crates/router/src/core/payment_methods/vault.rs

```diff
@@ -349,7 +358,15 @@ impl Vault {
 
         let lookup_key = token_id.unwrap_or_else(|| generate_id_with_default_len("token"));
 
-        create_tokenize(state, value1, Some(value2), lookup_key).await
+        let lookup_key = create_tokenize(state, value1, Some(value2), lookup_key).await?;
+        add_tokenize_data_task(
+            &*state.store,
+            &lookup_key,
+            pm,
+            "Delete_Tokenize_Data_Workflow",
```

Comment:
use snake_case or pascal_case, avoid mixing both

Reply:
If different tasks going to come for `tokenized_data` create a enum with a variant `Delete_Tokenize_Data_Workflow`. If this will always only one task then prefer adding the task name in the function itself.

Reply:
it was runner moved to appropriate place